tests/run/cpp_stl_random.pyx

```diff
@@ -53,3 +62,305 @@ def mt19937_discard(z):
     for _ in range(z + 1):
         b = rd()
     return a, b
+
+
+ctypedef fused any_dist:
+    uniform_int_distribution[int]
+    uniform_real_distribution[double]
+    bernoulli_distribution
+    binomial_distribution[int]
+    geometric_distribution[int]
+    negative_binomial_distribution[int]
+    poisson_distribution[int]
+    exponential_distribution[double]
+    gamma_distribution[double]
+    weibull_distribution[double]
+    extreme_value_distribution[double]
+    normal_distribution[double]
+    lognormal_distribution[double]
+    chi_squared_distribution[double]
+    cauchy_distribution[double]
+    fisher_f_distribution[double]
+    student_t_distribution[double]
+
+
+cdef sample_or_range(any_dist dist, bint sample):
+    """
+    This helper function returns a sample if `sample` is truthy and the range of the distribution
+    if `sample` is falsy. We use a fused type to avoid duplicating the conditional statement in each
+    distribution test.
+    """
+    cdef random_device rd
+    if sample:
+        return dist(mt19937(rd()))
+    else:
+        return dist.min(), dist.max()
+
+
+def uniform_int_distribution_test(a, b, sample=True):
+    """
+    >>> uniform_int_distribution_test(0, 0)
+    0
+    >>> uniform_int_distribution_test(0, 1) < 2
+    True
+    >>> uniform_int_distribution_test(5, 9, False)
+    (5, 9)
+    """
+    cdef uniform_int_distribution[int] dist = uniform_int_distribution[int](a, b)
+    return sample_or_range[uniform_int_distribution[int]](dist, sample)
+
+
+def uniform_real_distribution_test(a, b, sample=True):
+    """
+    >>> uniform_real_distribution_test(0, 0)
+    0.0
+    >>> x = uniform_real_distribution_test(0, 5)
+    >>> 0 < x and x < 5
+    True
+    >>> uniform_real_distribution_test(3, 8, False)
+    (3.0, 8.0)
+    """
+    cdef uniform_real_distribution[double] dist = uniform_real_distribution[double](a, b)
+    return sample_or_range[uniform_real_distribution[double]](dist, sample)
+
+
+def bernoulli_distribution_test(proba, sample=True):
+    """
+    >>> bernoulli_distribution_test(0)
+    False
+    >>> bernoulli_distribution_test(1)
+    True
+    >>> bernoulli_distribution_test(0.7, False)
+    (False, True)
+    """
+    cdef bernoulli_distribution dist = bernoulli_distribution(proba)
+    return sample_or_range[bernoulli_distribution](dist, sample)
+
+
+def binomial_distribution_test(n, proba, sample=True):
+    """
+    >>> binomial_distribution_test(10, 0)
+    0
+    >>> binomial_distribution_test(10, 1)
+    10
+    >>> x = binomial_distribution_test(100, 0.5)
+    >>> 30 < x and x < 70 or x  # Passes with high probability (>99.9%).
```

Comment:
Just for my understanding: the purpose of `or x` or these tests is to change the test into something that always passes? So we're just testing that you can call the distribution really? (Which is good, because I think it's the bit that we care about in Cython)